client/components/favorite-display.jsx

```diff
@@ -0,0 +1,131 @@
+  }
+
+  detailView(event) {
+    if (this.state.isViewing) {
+      this.setState({ isViewing: '' });
+    } else {
+      this.setState({ isViewing: event.target.getAttribute('data-id') });
+    }
+  }
+
+  handleSave() {
+    const token = window.localStorage.getItem('react-context-jwt');
+    fetch('/api/favorite/savedFavorite', {
+      method: 'POST',
+      headers: {
+        'react-context-jwt': token,
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(this.props.searchResult)
+    })
+      .then(response => response.json())
+      .catch(error => {
+        console.error('error', error);
+      });
+  }
+
+  render() {
+    if (!this.props.searchResult) {
+      return (
+        <div className="no-result-message-holder">
+          <h1 className="no-result-message">Sorry, No Result is Showing...</h1>
+        </div>
+      );
+    }
+    if (!this.props.searchResult.idAlbum) {
+      const {
+        strArtistThumb, strArtist, intBornYear, strGenre,
+        strStyle, strCountry, strWebsite, strArtistBanner, strBiographyEN
+      } = this.props.searchResult;
+      return (
+      <>
+        <div className="single-search col-nine-tenth">
+          <div className="default-row col-full">
+            <div className="col-half">
+              <div className="search-pic-holder">
+                  <img src={strArtistThumb} />
+              </div>
+            </div>
+            <div className="col-half search-info-col">
+              <div className="result-button-row">
+                  <button onClick={this.detailView} data-id={strArtist} type="button"
+                    className={this.state.isViewing === strArtist ? 'detail-clicked' : 'detail-button'}>Detail</button>
+                <button onClick={this.handleSave} className="add-button"><i className="fas fa-plus"></i></button>
+              </div>
+                <h3 className="caro-text">Artist: <span>{strArtist}</span></h3>
+                <h3 className="caro-text">Born: <span>{intBornYear || 'N/A'}</span></h3>
+                <h3 className="caro-text">Genre: <span>{strGenre || 'N/A'}</span></h3>
+                <h3 className="caro-text">Style: <span>{strStyle || 'N/A'}</span></h3>
+                <h3 className="caro-text">Origin: <span>{strCountry || 'N/A'}</span></h3>
+                <h3 className="caro-text">Artist Website: <span>{strWebsite || 'N/A'}</span></h3>
+            </div>
+          </div>
+            <div className={this.state.isViewing === strArtist ? 'hidden-detail-row col-full' : 'hidden-detail-row col-full hidden'}>
+            <div className="banner-holder col-full">
+                <img src={strArtistBanner} />
+            </div>
+            <div className="search-note col-full">
+              <h3 className="caro-text">Artist Biography:</h3>
+              <p className="detail-note">
+                  {strBiographyEN}
+              </p>
+            </div>
+          </div>
+        </div>
+      </>
```

Comment:
Similar to the carousel slide, when you have 30+ lines in `if/else` statements and a lot is similar, try to make a component that does the repeat elements and shows children. (like PageContainer)